db/version_builder.cc

```diff
@@ -1020,27 +1043,17 @@ class VersionBuilder::Rep {
 
   void MaybeAddFile(VersionStorageInfo* vstorage, int level, FileMetaData* f) {
     const uint64_t file_number = f->fd.GetNumber();
-
     const auto& level_state = levels_[level];
+    const auto& del_files = level_state.deleted_base_files;
 
-    const auto& del_files = level_state.deleted_files;
-    const auto del_it = del_files.find(file_number);
-
-    if (del_it != del_files.end()) {
-      // f is to-be-deleted table file
-      vstorage->RemoveCurrentStats(f);
```

Comment:
In the old logic, where is the stats being added (so we need to remove them here)? Just want to understand the logic.

Reply:
I actually have no idea why we previously `RemoveCurrentStats` when the file is not yet added to the new vstorage. Was hoping someone with more expertise could help explain this part.

Reply:
Looks like it was added in https://github.com/facebook/rocksdb/commit/355fa94365b4cf8b92938bee0becb18bc4526d82 but no explanation of how it works. Can you add a simple test to verify we shouldn't remove stats here?

Reply:
It seems version storage stats are passed down during initialization, that's why we need to remove stats of deleted files. I'll fix this and verify the key estimation is unchanged after moving files.